eng/tools/typespec-validation/src/rules/format.ts

```diff
@@ -10,9 +11,17 @@ export class FormatRule implements Rule {
     // Format parent folder to include shared files
 
     let [err, stdOutput, errorOutput] = await host.runCmd(`npx tsp format "../**/*.tsp"`, folder);
+
+    const gitDiffResult = await gitDiffTopSpecFolder(host, folder);
```

Comment:
Don't call "git diff" if "tsp compile" or "tsp format" fails. The result will probably be misleading or useless.